src/ui/button/index.tsx

```diff
@@ -0,0 +1,50 @@
+
+import React, { FC, useState } from 'react';
+import SvgGmail from '../../assets/iconComponents/Gmail'
+
+import { Button, ButtonProps } from '@mui/material';
+
+interface IProps extends ButtonProps {
+    buttonType?: string,
+    buttonInput?: string,
+    onclick?: Function
+}
+
+const ButtonUi: FC<IProps> = ({
+  variant,
+  color,
+  children,
+  size,
+  disabled,
+  onclick,
+  buttonInput,
+  buttonType = '',
+  ...rest
+}) => {
+
+    const [btnState, setBtnState] = useState<IProps>({
+        buttonType: 'login',
+        buttonInput: 'SUBMIT'
+    })
+
+    const renderBtn = () => {
+        if (btnState.buttonType === 'login') {
+            return <Button onClick={() => onclick} variant='contained' color='primary' size={size} disabled={disabled} sx={{ bgcolor: 'linear-gradient(to right, $blue-700, $blue-200)', '&:hover': { bgcolor: 'linear-gradient(to right, $blue-700, $blue-200)'} }}>Login</Button>
```

Comment:
Don't hard code `disabled`, by default the value is undefined and will turn into false, the logic is not misplaced here.

If you just need to pass down the value without modification, then use `...rest`.

Reply:
Removed hardcoded disabled.